src/App.js

```diff
@@ -1,16 +1,39 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
-
+import ChatEntry from './components/ChatEntry';
+import TimeStamp from './components/TimeStamp';
+import ChatLog from './components/ChatLog';
+import { useState } from 'react';
 const App = () => {
+  const [chatMessagesData, setChatMessagesData] = useState(chatMessages);
+
+  const onLike = (id) => {
+    setChatMessagesData((chatMessagesData) =>
+      chatMessagesData.map((message) => {
+        if (message.id === id) {
+          return { ...message, liked: !message.liked };
+        } else {
+          return message;
+        }
+      })
+    );
+  };
+  const calcTotalLikes = (chatMessagesData) => {
+    return chatMessagesData.reduce((total, message) => {
+      return message.liked ? (total += 1) : total;
```

Comment:
This does work but only because the assignment `total += 1` happens to return the new value. If it didn't, you'd have a bug. You don't need to reassign this variable at all, you can simply return `total + 1`